airbyte-integrations/connectors/source-youtube-analytics/source_youtube_analytics/source.py

```diff
@@ -0,0 +1,183 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+
+import csv
+import datetime
+import io
+import json
+import pkgutil
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream, HttpSubStream
+from airbyte_cdk.sources.streams.http.requests_native_auth import Oauth2Authenticator
+from airbyte_cdk.sources.utils.transform import TransformConfig, TypeTransformer
+
+
+class JobsStream(HttpStream):
+    " https://developers.google.com/youtube/reporting/v1/reference/rest/v1/jobs "
+
+    name = None
+    primary_key = None
+    http_method = None
+    url_base = "https://youtubereporting.googleapis.com/v1/"
+    JOB_NAME = "Airbyte reporting job"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        return [response.json()]
+
+    def path(self, **kwargs) -> str:
+        return "jobs"
+
+    def request_body_json(self, **kwargs) -> Optional[Mapping]:
+        if self.name:
+            return {"name": self.JOB_NAME, "reportTypeId": self.name}
+
+    def list(self):
+        " https://developers.google.com/youtube/reporting/v1/reference/rest/v1/jobs/list "
+        self.name = None
+        self.http_method = "GET"
+        results = list(self.read_records(sync_mode=None))
+        result = results[0]
```

Comment:
check if result is not empty

Reply:
yeah, we have already talked about this with @bazarnov 
it seems it's not very obvious for this stream class ( not sure how to make it more obvious )
but this stream always returns one and only one result because 
as you can see `parse_response` returns one and only one result.